libnetwork/endpoint.go

```diff
@@ -595,36 +595,103 @@ func (ep *Endpoint) sbJoin(ctx context.Context, sb *Sandbox, options ...Endpoint
 		sb.resolver.SetForwardingPolicy(sb.hasExternalAccess())
 	}
 
-	currentExtEp := sb.getGatewayEndpoint()
-	moveExtConn := currentExtEp != extEp
-	if moveExtConn {
-		if extEp != nil {
-			log.G(ctx).Debugf("Revoking external connectivity on endpoint %s (%s)", extEp.Name(), extEp.ID())
-			extN, err := extEp.getNetworkFromStore()
-			if err != nil {
-				return fmt.Errorf("failed to get network from store for revoking external connectivity during join: %v", err)
+	gwepAfter4, gwepAfter6 := sb.getGatewayEndpoint()
```

Comment:
Heck, not critizing your code, but that level of Single Responsibility violation is super crazy! 🙈 

We should really straighten that code up at some point.